src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -59,7 +68,42 @@
   public List<IexHistoricalPrices> getHistoricalPricesForSymbols(final String symbols,
       final String range, final String date) {
 
-    return iexHistoricalClient.getHistoricalPricesForSymbols(symbols, range, date);
+    String dateFormatted = date.substring(0,4) + "-" + date.substring(4, 6)
+        + "-" + date.substring(6, 8);
+
+    HistoricalPricesId historicalPricesId =
+        new HistoricalPricesId(symbols, dateFormatted, "10:50");
+
+    if (historicalPricesService.exists(historicalPricesId)) {
+
+      List<IexHistoricalPrices> listIexHistoricalPrices = new ArrayList();
```

Comment:
adding prefixes like ```list``` and ```obj``` to variables is a code smell in Java, it should be easy to tell what the type of the object is due to the strong typing.
```ArrayList``` should be parameterized with ```IexHistoricalPrices```. Is this giving you a warning? Normally it would look like ```new ArrayList<IexHistoricalPrices>();``` or ```new ArrayList<>();``` for short.